doc/site/zh-cn/guides/thread-primer/node-roles-and-types.md

```diff
@@ -0,0 +1,125 @@
+# 节点角色和类型
+
+## 转发角色
+
+<figure class="attempt-right">
+<a href="../images/ot-primer-roles_2x.png"><img src="../images/ot-primer-roles.png" srcset="../images/ot-primer-roles.png 1x, ../images/ot-primer-roles_2x.png 2x" border="0" alt="OT Node Roles" /></a>
+</figure>
+
+在 Thread 网络中，节点分成两种转发角色：Router 和 End Device。
+
+### Router
+
+Router 节点的行为如下：
+
+* 为网络设备转发数据包
+* 为尝试加入网络的设备提供安全的 commissioning 服务
+* 始终打开它的收发器
+
+### End Device
+
+End Device 节点的行为如下：
+
+* 主要与单个 Router 进行通信
+* 不会为其他网络设备转发数据包
+* 可以关闭它的收发器来降低功耗
+
+Key Point: Router 和 End Device 之间的关系称为父子关系。End Device 正确地依附到一个 Router 上，其中 Router 始终作为父节点，End Device 则始终是子节点。
+
+## 设备类型
+
+此外，节点有许多种类型。
+
+<figure class="attempt-right">
+<a href="../images/ot-primer-taxonomy_2x.png"><img src="../images/ot-primer-taxonomy.png" srcset="../images/ot-primer-taxonomy.png 1x, ../images/ot-primer-taxonomy.png 2x" border="0" alt="OT Device Taxonomy" /></a>
+</figure>
+
+### Full Thread Device
+
+一个 FTD（Full Thread Device）总是打开它的无线电收发器，它订阅所有 Router 的多播地址，并维护 IPv6 地址映射。FTD 有三种类型：
+
+* Router
+* REED（Router Eligible End Device）— 可以升级为 Router
+* FED（Full End Device）— 无法升级为 Router
+
+FTD 可以作为 Router（父）或 End Device（子）。
+
+### Minimal Thread Device
+
+MTD（Minimal Thread Device）不会订阅多播流量，并且它会将它的所有消息发送给它的父节点。MTD 有两种类型：
+
+* MED（Minimal End Device）— 始终打开自身的收发器，无需从父节点中轮询消息
+* SED（Sleepy End Device）— 通常会关闭自身的收发器（睡眠），偶然会打开收发器（唤醒）以从父节点中轮询消息
+
+MTD 只能作为 End Device（子）。
+
+### 升级和降级
+
+当一个 REED 是希望加入到 Thread 网络的新 End Device 所能到达的唯一节点时，它可以升级成 Router：
```

Comment:
```suggestion
当待加入设备仅能与 REED 通信时, 该 REED 可以升级成为 Router：
```

Reply:
Thread will make the number of routers between 16 and 23 (maximum 32).
Consider the following two situations:

1. The number of routers in the network is 23. At this time, there is a device that wants to join the network, and the device that can communicate with it is REED. Will this REED be upgraded?
2. If the number of routers is 32, what is the result?

Reply:
Good question.
for 1, it will.
for 2, it won't become a router successfully.

For existing attach process ([HandleParentRequest](https://github.com/openthread/openthread/blob/2ebdf874eb874ccb2d30c96c171878f1ce607c17/src/core/thread/mle_router.cpp#L1597-L1636)), I am thinking one additional rule could be introduced for enhancement:
> REED doesn't respond parent request if its partition already have 32 active routers as it is for sure that Leader would reject its address solicit.

Thoughts? @jwhui

Reply:
@librasungirl , makes sense to me!